### PR TITLE
Implement NewsCard component for NewsOverlay

### PR DESCRIPTION
**Pull request by EVAST9919**
Screenshot_1

### CODE REVIEW

File: osu.Game/Online/API/Requests/Responses/APINewsPost.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        public string EditUrl { get; set; }
+
+        [JsonProperty(@"first_image")]
+        public string FirstImage { get; set; }
+
+        [JsonProperty(@"published_at")]
+        public DateTime PublishedAt { get; set; }
+
+        [JsonProperty(@"updated_at")]
+        public DateTime UpdatedAt { get; set; }
```
Comment (bdach) [inline osu.Game/Online/API/Requests/Responses/APINewsPost.cs:27]: Pretty sure these should be `DateTimeOffset`s.

File: osu.Game/Online/API/Requests/Responses/APINewsPost.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// See the LICENCE file in the repository root for full licence text.
+
+using Newtonsoft.Json;
+using System;
+
+namespace osu.Game.Online.API.Requests.Responses
+{
+    public class APINewsPost
+    {
+        [JsonProperty(@"id")]
```
Comment (bdach) [inline osu.Game/Online/API/Requests/Responses/APINewsPost.cs:11]: Raw string prefixes seem unnecessary.

File: osu.Game/Overlays/News/NewsCard.cs [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+                if (source.StartsWith('/'))
+                    return "https://osu.ppy.sh" + source;
+
+                return source;
+            }
+        }
+
+        private class DateContainer : CircularContainer, IHasTooltip
+        {
+            public string TooltipText => date.ToString("d MMMM yyyy hh:mm:ss UTCz");
```
Comment (bdach) [inline osu.Game/Overlays/News/NewsCard.cs:161]: This tooltip is not consistent with web design. This should be using whatever `DrawableDate` is, probably by hoisting out `DrawableDate.DateTooltip` of the inner class and making it public.

File: osu.Game/Online/API/Requests/Responses/APINewsPost.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using Newtonsoft.Json;
+using System;
+
+namespace osu.Game.Online.API.Requests.Responses
+{
+    public class APINewsPost
```
Comment (bdach) [inline osu.Game/Online/API/Requests/Responses/APINewsPost.cs:9]: Where did you get this endpoint contract structure from? I looked at the news page on web and I can't find a straight API request that would just fetch this as JSON, [API docs are missing](https://osu.ppy.sh/docs/index.html#apiv2newsnews), and this PR does not have a web request attached that I can inspect.

**Review by bdach — CHANGES_REQUESTED** [review]

File: osu.Game/Online/API/Requests/Responses/APINewsPost.cs (hunk 4 above)
Comment (EVAST9919) [reply osu.Game/Online/API/Requests/Responses/APINewsPost.cs:9]: I just sent a raw request and checked the response
Screenshot_1

Comment (bdach) [reply osu.Game/Online/API/Requests/Responses/APINewsPost.cs:9]: I notice an `&amp;` HTML entity in the `author` field of post with ID 802 (I'm betting it's possible for `title` and `preview` to contain HTML entities, too). This should be handled at some point using `WebUtility.HtmlDecode` - probably best at deserialisation time.

Comment (EVAST9919) [reply osu.Game/Online/API/Requests/Responses/APINewsPost.cs:9]: Nice catch will consider!
Also, as a side note: web can't handle this either, should probably open an issue
Аннотация 2020-07-06 232512

Comment (bdach) [reply osu.Game/Online/API/Requests/Responses/APINewsPost.cs:9]: Opened ppy/osu-web#6274.

File: osu.Game/Graphics/DrawableDate.cs [hunk 5]
```diff
@@ -82,7 +82,7 @@ private void updateTimeWithReschedule()
 
         public object TooltipContent => Date;
 
-        private class DateTooltip : VisibilityContainer, ITooltip
+        public class DateTooltip : VisibilityContainer, ITooltip
```
Comment (smoogipoo) [inline osu.Game/Graphics/DrawableDate.cs:85]: Would like this to be moved to a separate file.

**Review by smoogipoo — CHANGES_REQUESTED** [review]
Looks fine otherwise

**Comment by bdach** [comment]
I've adjusted one font size to match web; otherwise than that looks good.

Going to make and exception and approve this despite having pushed to this PR as the change was menial.

**Review by bdach — APPROVED** [review]

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**